### DISCUSSION TITLE
Unable to run the command `forge build --zksync`

### DISCUSSION THREAD

**Discussion by PrashikshitSaini** [Cyfrin/foundry-full-course-cu#1907] [Q&A]
When I try to run the `forge build --zksync` , it throws the following error:
``` bash
[⠊] Compiling (zksync)...
Error: 
Broken pipe (os error 32)
```
There have been issue with some other users too, does anyone has the solution for the same?

**Comment by EngrPips** [comment]
Honestly, I haven't seen any solution yet, but hoping someone suggests a solution.

**Reply by EngrPips** [reply]
Hey @PrashikshitSaini, Someone mentioned cyfrin discord that they ran the command with Ubuntu 22.04, which works fine.

<img width="1440" alt="Screenshot 2024-06-30 at 05 24 26" src="https://github.com/Cyfrin/foundry-full-course-cu/assets/95760918/a40d5514-a75e-4a94-9492-7f8bd34e1141">

**Reply by PrashikshitSaini** [reply]
Thank You for replying @EngrPips , but I do have Ubuntu 22.04
```bash
root@Prashikshit:~# cat /etc/os-release
PRETTY_NAME="Ubuntu 22.04.4 LTS"
NAME="Ubuntu"
VERSION_ID="22.04"
VERSION="22.04.4 LTS (Jammy Jellyfish)"
VERSION_CODENAME=jammy
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=jammy
```

**Reply by a1111198** [reply]
I am also getting same Error and here is Ubuntu version.
```
Distributor ID: Ubuntu
Description:    Ubuntu 22.04.3 LTS
Release:        22.04
Codename:       jammy
```

**Reply by EngrPips** [reply]
I am unsure why it worked for them, but it is not for you guys. Thats weird

**Reply by RanggaAlroy** [reply]
Have you guys found the solution? I have the same problem too.

**Reply by PrashikshitSaini** [reply]
I just tried something, you can check the answer now(and correct me if I'm wrong)

**Comment by PrashikshitSaini** [comment]
So I am not clear sure please correct this if it's wrong!

So I first ran the ```forge build --zksync --help```
and this gave me the following output with this option:
```bash

      --zk-detect-missing-libraries
          Detect missing libraries, instead of erroring

          Currently unused
```
under the Zksync configuration. Well it says Currently unused but I ran the following command and it worked for me:
```bash
root@Prashikshit:~/Foundry/SimpleStorage# forge build --zk-detect-missing-libraries
[⠊] Compiling...
[⠊] Compiling 28 files with 0.8.26
[⠒] Solc 0.8.26 finished in 872.45ms
Compiler run successful!
```

**Reply by EngrPips** [reply]
This seems quite interesting. So you break free from this issue already?

**Reply by worca333** [reply]
Ironically, it worked as I changed my solc version to 0.8.25 in **foundry.toml**

```
solc="0.8.25"
```

**Reply by EngrPips** [reply]
It is interesting that we are breaking free from this issue already.

**Reply by RanggaAlroy** [reply]
oh great thank you @worca333 it works now!

**Reply by PrashikshitSaini** [reply]
> Ironically, it worked as I changed my solc version to 0.8.25 in **foundry.toml**
> 
> ```
> solc="0.8.25"
> ```

Doesn't it give warnings when run(although compiling)?

**Reply by PrashikshitSaini** [reply]
> This seems quite interesting. So you break free from this issue already?

More like escaping.

**Reply by EngrPips** [reply]
I mean, escaping is also a form of breaking free, haha.

**Reply by worca333** [reply]
> > Ironically, it worked as I changed my solc version to 0.8.25 in **foundry.toml**
> > ```
> > solc="0.8.25"
> > ```
> 
> Doesn't it give warnings when run(although compiling)?

Likely yes as it does in Patricks' course.
I believe the warning comes from the config of local chain env.

**Reply by EngrPips** [reply]
Interesting.
